docker/pulsar/Dockerfile

```diff
@@ -58,10 +58,20 @@ RUN sed -i "s|http://archive\.ubuntu\.com/ubuntu/|${UBUNTU_MIRROR:-mirror://mirr
      && echo 'Acquire::http::Timeout "30";\nAcquire::ftp::Timeout "30";\nAcquire::Retries "3";' > /etc/apt/apt.conf.d/99timeout_and_retries \
      && apt-get update \
      && apt-get -y dist-upgrade \
-     && apt-get -y install --no-install-recommends openjdk-17-jdk-headless netcat dnsutils less procps iputils-ping \
+     && apt-get -y install --no-install-recommends netcat dnsutils less procps iputils-ping \
                  python3 python3-kazoo python3-pip \
-                 curl ca-certificates \
-     && apt-get -y --purge autoremove \
+                 curl ca-certificates wget apt-transport-https
+
+# Install Eclipse Temurin Package
+RUN mkdir -p /etc/apt/keyrings \
+     && wget -O - https://packages.adoptium.net/artifactory/api/gpg/key/public | tee /etc/apt/keyrings/adoptium.asc \
+     && echo "deb [signed-by=/etc/apt/keyrings/adoptium.asc] https://packages.adoptium.net/artifactory/deb $(awk -F= '/^VERSION_CODENAME/{print$2}' /etc/os-release) main" | tee /etc/apt/sources.list.d/adoptium.list \
+     && apt-get update \
+     && apt-get -y dist-upgrade \
+     && apt-get -y install temurin-17-jdk
+
+# Cleanup apt
+RUN apt-get -y --purge autoremove \
```

Comment:
Don't we need to run this in the same command that we run our `apt-get` installations in order to get the benefit of a smaller docker image? Otherwise, we'll add files in one layer and remove them in another.

Reply:
Good point. Would you like to prepare a patch and show the difference in size?

Reply:
Included in https://github.com/apache/pulsar/pull/17733. You can give it a review :)

Reply:
Revert as if I perform this command merge, the build will fail with `returned a non-zero code: 100`.